Ft sprint3 admin site

helloCaller opened the pull request:
This PR includes a refactor of focusedOccasion to follow the logic happening with focusedEvent. Previously using dispatch, a static state of focusedOccasion would get passed around, but now all components have access to focusedOccasion object as it changes on the server. This has made a separate occasion state redundant, as UI can directly get affected by the state of focusedOccasion (opened, or closed).

helloCaller committed:
refactor of focused Occasion

helloCaller committed:
clode cleaning

helloCaller committed:
updated readme, made post function, small code review

jakemoves commented on admin-site/src/Occasion.svelte:
```diff
@@ -6,159 +6,113 @@
+  }
+
+  function openOccasionButton() {
+    serverCallToOpenOrCloseOccasion("opened"); 
```
How about `updateOccasionStateOnServer` as a function name?

helloCaller replied:
Implemented

jakemoves commented on admin-site/src/Occasion.svelte:
```diff
@@ -6,159 +6,113 @@
+  };
+
+
+  function deleteOccasion() {
```
I know we have at least one error case that relates here — if the occasion is open it can't be deleted. Does this get handled anywhere?

helloCaller replied:
While the occasion is open, currently there is no option to delete the occasion in the UI (only in its closed state). If the user tries to delete the event with an open occasion, that throws an error telling them which occasion is open.

jakemoves commented on admin-site/src/Occasion.svelte:
```diff
@@ -6,159 +6,113 @@
+  let formattedEndTime;
+
+  //double check that a delete Occasion has happened
+  let deleteOccasionHasHappened = false;
```
Where does this get used?

helloCaller replied:
It doesn't anymore. Removed.

jakemoves commented on admin-site/src/Occasion.svelte:
```diff
@@ -204,56 +158,54 @@ function deleteOccasion() {
 
 
+<!-- update UI based on state of occasion -->
+{#if focusedOccasion.state == "closed"}
```
Lovely.

jakemoves commented on admin-site/src/OccasionsList.svelte:
```diff
@@ -54,30 +45,18 @@
 
   function occasionButton(id) {
```
I don't have a problem with this, but it would be sliiiightly more common / customary to keep the markup super simple (i.e. just let on:click call its handler and pass the usual `DOMEvent` object automatically), and then in your handler function, grab `event.target.id`. (https://developer.mozilla.org/en-US/docs/Web/API/Event/target)

helloCaller replied:
I had to use event.currentTarget.value. I believe that's because we have the Button component calling the event.

jakemoves reviewed (commented):
Looks good. I think you should write a short (paragraph or three, or bullet points) explanation of the flow you've created using stores, for documentation. We can decide where's best for it to live later.

helloCaller committed:
PR requested changes

jakemoves commented:
Derp, of course. Maybe drop a comment to that effect in there, where the error handling would otherwise go. 

> On Feb 25, 2020, at 12:22 AM, helloCaller <notifications@github.com> wrote:
> 
> ﻿
> @helloCaller commented on this pull request.
> 
> In admin-site/src/Occasion.svelte:
> 
> >    };
>  
> - function closeOccasionButton(){
> +  function closeOccasionButton(){
> +    serverCallToOpenOrCloseOccasion("closed");  
> +  };
> +
> +
> +  function deleteOccasion() {
> While the occasion is open, currently there is no option to delete the occasion in the UI (only in its closed state). If the user tries to delete the event with an open occasion, that throws an error telling them which occasion is open.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

helloCaller committed:
event.label title added, fix for long event names, title cased buttons, reduced space under titles, added enumerators for media info

helloCaller committed:
renamed pageStore to UpdateUIstore

helloCaller committed:
added file for error messages, subheading tweeks, added event id

helloCaller commented:
In case it helps a quick run down of what was done since this initial PR.
- added event.label and event.id names in heading of occasions list 
- updated delete buttons to contain names/labels
- fixed spacing for case in which the title of the event is very long 
- title cased buttons that have Event or Occasion 
- added enumerators for media
- added file for error messages 
- edited css on sub headings

helloCaller committed:
added note about occasion delete, implemented some template literals for legibility

jakemoves commented on admin-site/src/ErrorMessages.js:
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  formEmptyNameField : `Please enter a label/name above`
```
Please remove 'label/', let's just use 'name'

helloCaller replied:
Same for the label on the actual field? I put both cause in user terms it's "name", but we have it as label in developer terms (in the code).

jakemoves commented on admin-site/src/Occasion.svelte:
```diff
@@ -32,11 +32,33 @@
-
   let cueState = 0;
 
+  let mediaDomainEnum = {
```
Declaring this with `const` rather than `let` would be a clearer signal of intent.

helloCaller replied:
Implemented

jakemoves commented on admin-site/src/Occasion.svelte:
```diff
@@ -32,11 +32,33 @@
+    5:"Image"
+  };
+
+  let cueActionEnum = {
```
Same here, see if `const` works.

jakemoves commented on admin-site/src/Occasion.svelte:
```diff
@@ -60,31 +82,33 @@
 
 
   function deleteOccasion() {
+    //Delete Occasion is only available in the UI when an occasion is closed
```
This is good but not quite the right amount of context. 

"Server will throw an error if we try to delete an open occasion, but we only show the Delete Occasion button when an occasion is closed, so we don't need to handle showing that error to users." (or something like that)

helloCaller replied:
Implemented

jakemoves commented:
Yeah, I think name makes more sense for users so please change it in both

> On Feb 26, 2020, at 12:03 PM, helloCaller <notifications@github.com> wrote:
> 
> ﻿
> @helloCaller commented on this pull request.
> 
> In admin-site/src/ErrorMessages.js:
> 
> > @@ -0,0 +1,3 @@
> +module.exports = {
> +  formEmptyNameField : `Please enter a label/name above`
> Same for the label on the actual field? I put both cause in user terms it's "name", but we have it as label in developer terms (in the code).
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

helloCaller committed:
PR tweeks, changed label to name, added comments, const to enums
